solidity/contracts/vault/TBTCOptimisticMinting.sol

```diff
@@ -0,0 +1,557 @@
+    //          will be the value of the delay.
+    uint32 public newOptimisticMintingDelay;
+    /// @notice The timestamp at which the update of the optimistic minting
+    ///         delay started. Zero if update is not in progress.
+    uint256 public optimisticMintingDelayUpdateInitiatedTimestamp;
+
+    event OptimisticMintingRequested(
+        address indexed minter,
+        uint256 indexed depositKey,
+        address indexed depositor,
+        uint256 amount,
+        bytes32 fundingTxHash,
+        uint32 fundingOutputIndex
+    );
+    event OptimisticMintingFinalized(
+        address indexed minter,
+        uint256 indexed depositKey,
+        address indexed depositor,
+        uint256 optimisticMintingDebt
+    );
+    event OptimisticMintingCancelled(
+        address indexed guardian,
+        uint256 indexed depositKey
+    );
+    event OptimisticMintingDebtRepaid(
+        address indexed depositor,
+        uint256 optimisticMintingDebt
+    );
+    event MinterAdded(address indexed minter);
+    event MinterRemoved(address indexed minter);
+    event GuardianAdded(address indexed guardian);
+    event GuardianRemoved(address indexed guardian);
+    event OptimisticMintingPaused();
+    event OptimisticMintingUnpaused();
+
+    event OptimisticMintingFeeUpdateStarted(
+        uint32 newOptimisticMintingFeeDivisor
+    );
+    event OptimisticMintingFeeUpdated(uint32 newOptimisticMintingFeeDivisor);
+
+    event OptimisticMintingDelayUpdateStarted(uint32 newOptimisticMintingDelay);
+    event OptimisticMintingDelayUpdated(uint32 newOptimisticMintingDelay);
+
+    modifier onlyMinter() {
+        require(isMinter[msg.sender], "Caller is not a minter");
+        _;
+    }
+
+    modifier onlyGuardian() {
+        require(isGuardian[msg.sender], "Caller is not a guardian");
+        _;
+    }
+
+    modifier onlyOwnerOrGuardian() {
+        require(
+            owner() == msg.sender || isGuardian[msg.sender],
+            "Caller is not the owner or guardian"
+        );
+        _;
+    }
+
+    modifier whenOptimisticMintingNotPaused() {
+        require(!isOptimisticMintingPaused, "Optimistic minting paused");
+        _;
+    }
+
+    modifier onlyAfterGovernanceDelay(uint256 updateInitiatedTimestamp) {
+        GovernanceUtils.onlyAfterGovernanceDelay(
+            updateInitiatedTimestamp,
+            GOVERNANCE_DELAY
+        );
+        _;
+    }
+
+    constructor(Bridge _bridge) {
+        require(
+            address(_bridge) != address(0),
+            "Bridge can not be the zero address"
+        );
+
+        bridge = _bridge;
+    }
+
+    /// @dev Mints the given amount of TBTC to the given depositor's address.
+    ///      Implemented by TBTCVault.
+    function _mint(address minter, uint256 amount) internal virtual;
+
+    /// @notice Allows to fetch a list of all Minters.
+    function getMinters() external view returns (address[] memory) {
+        return minters;
+    }
```

Comment:
It is needed so that we can fetch all minters in one call with `await tbtcVault.getMinters()`. Without this function, we'd have to fetch them index by index: `await tbtcVault.minters(0)`, `await tbtcVault.minters(1)`, ...